crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -138,9 +162,32 @@ fn is_first_party(qualified_name: &str, level: u32, checker: &Checker) -> bool {
     }
 }
 
+/// Find the `Expr` for top level `__all__` bindings.
+fn find_dunder_all_exprs<'a>(semantic: &'a SemanticModel) -> Vec<&'a ast::Expr> {
+    semantic
+        .global_scope()
+        .get_all("__all__")
+        .filter_map(|binding_id| {
+            let binding = semantic.binding(binding_id);
+            let stmt = match binding.kind {
+                BindingKind::Export(_) => binding.statement(semantic),
+                _ => None,
+            }?;
+            match stmt {
+                Stmt::Assign(ast::StmtAssign { value, .. }) => Some(&**value),
+                Stmt::AnnAssign(ast::StmtAnnAssign { value, .. }) => value.as_deref(),
+                Stmt::AugAssign(ast::StmtAugAssign { value, .. }) => Some(&**value),
+                _ => None,
+            }
+        })
```

Comment:
Possible optimisation here: if there's more than one `__all__` definition, we won't do an autofix anyway, so we only need to collect a maximum of two `__all__` definitions into the `Vec` being returned

```suggestion
        })
        .take(2)
```

Reply:
I'd probably vote to leave it as-is -- it makes sense (we can avoid some binding lookups), but seems unlikely to matter and creates some implicit coupling between this method and the fix rules that we may forget about in the future.

Reply:
Whoops. I clicked the button. I'll revert if it's not conclusive.

Reply:
I defer to you as the author, either is fine :)

Reply:
Interesting. No effect. :)) I'll revert.

Reply:
Yeah, no difference at all :( Ah, well.